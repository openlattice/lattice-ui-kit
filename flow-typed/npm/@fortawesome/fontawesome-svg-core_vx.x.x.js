// flow-typed signature: 39887e5e6c6c32861704232d9719cd14
// flow-typed version: <<STUB>>/@fortawesome/fontawesome-svg-core_v1.2.4/flow_v0.80.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@fortawesome/fontawesome-svg-core'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@fortawesome/fontawesome-svg-core' {
  declare module.exports: any;
}
