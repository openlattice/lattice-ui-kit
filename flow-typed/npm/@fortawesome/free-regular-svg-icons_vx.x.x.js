// flow-typed signature: e06b1e176fa03c6cedb9f03bb6cc7872
// flow-typed version: <<STUB>>/@fortawesome/free-regular-svg-icons_v5.13.0/flow_v0.118.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@fortawesome/free-regular-svg-icons'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@fortawesome/free-regular-svg-icons' {
  declare module.exports: any;
}
