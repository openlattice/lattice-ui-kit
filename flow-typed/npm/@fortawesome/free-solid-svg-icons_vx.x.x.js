// flow-typed signature: 64f61153e222e791ad6eca6c60c55baa
// flow-typed version: <<STUB>>/@fortawesome/free-solid-svg-icons_v5.13.0/flow_v0.118.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@fortawesome/free-solid-svg-icons'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@fortawesome/free-solid-svg-icons' {
  declare module.exports: any;
}
