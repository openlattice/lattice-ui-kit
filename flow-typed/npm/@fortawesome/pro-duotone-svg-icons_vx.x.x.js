// flow-typed signature: dc91210f6aed06e504e8c5d05af6775b
// flow-typed version: <<STUB>>/@fortawesome/pro-duotone-svg-icons_v5.10.2/flow_v0.106.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@fortawesome/pro-duotone-svg-icons'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@fortawesome/pro-duotone-svg-icons' {
  declare module.exports: any;
}
