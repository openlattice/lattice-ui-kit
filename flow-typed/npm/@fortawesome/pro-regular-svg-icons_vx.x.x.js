// flow-typed signature: c41877c2f635a0484c823342d488e66e
// flow-typed version: <<STUB>>/@fortawesome/pro-regular-svg-icons_v5.3.1/flow_v0.80.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@fortawesome/pro-regular-svg-icons'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@fortawesome/pro-regular-svg-icons' {
  declare module.exports: any;
}
