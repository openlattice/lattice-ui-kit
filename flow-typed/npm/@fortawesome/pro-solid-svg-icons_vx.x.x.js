// flow-typed signature: b2f94000db186d565d3eb121a020bf27
// flow-typed version: <<STUB>>/@fortawesome/pro-solid-svg-icons_v5.3.1/flow_v0.80.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@fortawesome/pro-solid-svg-icons'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@fortawesome/pro-solid-svg-icons' {
  declare module.exports: any;
}
