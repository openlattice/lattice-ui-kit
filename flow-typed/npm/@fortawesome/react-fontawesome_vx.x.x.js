// flow-typed signature: 561f463b72d39e8e0057aaad26de937b
// flow-typed version: <<STUB>>/@fortawesome/react-fontawesome_v0.1.2/flow_v0.80.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@fortawesome/react-fontawesome'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@fortawesome/react-fontawesome' {
  declare module.exports: any;
}
