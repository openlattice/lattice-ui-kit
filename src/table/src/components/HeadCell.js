// @flow
import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import type { Node } from 'react';
import { faSort, faSortUp, faSortDown } from '@fortawesome/pro-duotone-svg-icons';

import { Cell } from './styled';
import type { SortOrder } from '../../types';

type Props = {
  cellStyle ? :Object;
  children :Node;
  className ? :string;
  onClick ? :(e :SyntheticEvent<HTMLElement>) => void;
  order ? :SortOrder;
  sortable ? :boolean;
};

const HeadCell = (props :Props) => {
  const {
    cellStyle,
    children,
    className,
    onClick,
    order,
    sortable,
  } = props;

  let icon = faSort;
  if (order === 'asc') icon = faSortUp;
  if (order === 'desc') icon = faSortDown;

  const sortText = `Sort (${order === 'asc' ? 'Ascending' : 'Descending'})`;

  return (
    <Cell
        as="th"
        cellStyle={cellStyle}
        className={className}
        onClick={onClick}>
      {children}
      { (sortable) && <span aria-label={sortText}><FontAwesomeIcon icon={icon} fixedWidth /></span> }
    </Cell>
  );

};

HeadCell.defaultProps = {
  cellStyle: {},
  className: undefined,
  onClick: undefined,
  order: false,
  sortable: false,
};

export default HeadCell;
