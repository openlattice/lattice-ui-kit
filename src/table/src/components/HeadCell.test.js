import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { faSort, faSortUp, faSortDown } from '@fortawesome/pro-duotone-svg-icons';

import HeadCell from './HeadCell';
import { Cell } from '../..';

describe('HeadCell', () => {

  // shallow render to avoid invalid DOM error
  describe('props', () => {

    describe('onClick', () => {
      test('invoke onClick', () => {
        const mockOnClick = jest.fn();
        const wrapper = shallow(<HeadCell onClick={mockOnClick} />);

        expect(toJson(wrapper)).toMatchSnapshot();

        expect(mockOnClick).toHaveBeenCalledTimes(0);
        wrapper.find('Cell').simulate('click');
        expect(mockOnClick).toHaveBeenCalledTimes(1);
      });
    });

    describe('sortable', () => {
      test('should render FontAwesomeIcon if true', () => {
        const wrapper = shallow(<HeadCell sortable />);

        expect(wrapper.find('FontAwesomeIcon')).toHaveLength(1);
      });

      test('should not render FontAwesomeIcon if false', () => {
        const wrapper = shallow(<HeadCell />);

        expect(wrapper.find('FontAwesomeIcon')).toHaveLength(0);
      });
    });

    describe('order', () => {
      test('should render faSortUp if sortable and order="asc"', () => {
        const wrapper = shallow(<HeadCell sortable order="asc" />);

        expect(toJson(wrapper)).toMatchSnapshot();
        expect(wrapper.find('FontAwesomeIcon').prop('icon')).toEqual(faSortUp);
      });

      test('should render faSortDown if sortable and order="desc"', () => {
        const wrapper = shallow(<HeadCell sortable order="desc" />);

        expect(toJson(wrapper)).toMatchSnapshot();
        expect(wrapper.find('FontAwesomeIcon').prop('icon')).toEqual(faSortDown);
      });

      test('should render faSort if sortable but without order', () => {
        const wrapper = shallow(<HeadCell sortable />);

        expect(toJson(wrapper)).toMatchSnapshot();
        expect(wrapper.find('FontAwesomeIcon').prop('icon')).toEqual(faSort);
      });
    });
  });

  describe('render', () => {
    test('should render as th', () => {
      const wrapper = shallow(<HeadCell />);

      expect(wrapper.find(Cell).prop('as')).toEqual('th');
    });

    test('should render children', () => {
      const child = <div>child</div>;
      const wrapper = shallow(<HeadCell>{child}</HeadCell>);

      expect(wrapper.contains(child)).toEqual(true);
    });
  });
});
