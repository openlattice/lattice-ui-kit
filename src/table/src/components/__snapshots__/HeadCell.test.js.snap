// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeadCell props onClick invoke onClick 1`] = `
<Cell
  as="th"
  cellStyle={Object {}}
  onClick={[MockFunction]}
/>
`;

exports[`HeadCell props order should render faSort if sortable but without order 1`] = `
<Cell
  as="th"
  cellStyle={Object {}}
>
  <span>
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={true}
      flip={null}
      icon={
        Object {
          "icon": Array [
            320,
            512,
            Array [],
            "f0dc",
            Array [
              "M279.05 288.05h-238c-21.4 0-32.07 25.95-17 41l119.1 119 .1.1a23.9 23.9 0 0 0 33.8-.1l119-119c15.1-15.05 4.4-41-17-41zm-238-64h238c21.4 0 32.1-25.9 17-41l-119-119a.94.94 0 0 0-.1-.1 23.9 23.9 0 0 0-33.8.1l-119.1 119c-15.05 15.1-4.4 41 17 41z",
              "",
            ],
          ],
          "iconName": "sort",
          "prefix": "fad",
        }
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
  </span>
</Cell>
`;

exports[`HeadCell props order should render faSortDown if sortable and order="desc" 1`] = `
<Cell
  as="th"
  cellStyle={Object {}}
>
  <span>
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={true}
      flip={null}
      icon={
        Object {
          "icon": Array [
            320,
            512,
            Array [],
            "f0dd",
            Array [
              "M279.07 224.05h-238c-21.4 0-32.1-25.9-17-41l119-119a23.9 23.9 0 0 1 33.8-.1l.1.1 119.1 119c15.07 15.05 4.4 41-17 41z",
              "M296.07 329.05L177 448.05a23.9 23.9 0 0 1-33.8.1l-.1-.1-119-119c-15.1-15.1-4.4-41 17-41h238c21.37 0 32.04 25.95 16.97 41z",
            ],
          ],
          "iconName": "sort-down",
          "prefix": "fad",
        }
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
  </span>
</Cell>
`;

exports[`HeadCell props order should render faSortUp if sortable and order="asc" 1`] = `
<Cell
  as="th"
  cellStyle={Object {}}
>
  <span>
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={true}
      flip={null}
      icon={
        Object {
          "icon": Array [
            320,
            512,
            Array [],
            "f0de",
            Array [
              "M41.05 288.05h238c21.4 0 32.1 25.9 17 41l-119 119a23.9 23.9 0 0 1-33.8.1l-.1-.1-119.1-119c-15.05-15.05-4.4-41 17-41z",
              "M24.05 183.05l119.1-119A23.9 23.9 0 0 1 177 64a.94.94 0 0 1 .1.1l119 119c15.1 15.1 4.4 41-17 41h-238c-21.45-.05-32.1-25.95-17.05-41.05z",
            ],
          ],
          "iconName": "sort-up",
          "prefix": "fad",
        }
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
  </span>
</Cell>
`;
